gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/editor/GravatarQuickEditorParams.kt

```diff
@@ -0,0 +1,83 @@
+package com.gravatar.quickeditor.ui.editor
+
+import com.gravatar.quickeditor.ui.oauth.OAuthParams
+import java.util.Objects
+
+/**
+ * Quick Editor's configuration parameters
+ *
+ * @property appName The App Name that is launching the Gravatar Quick Editor
+ * @property oAuthParams The OAuth parameters for your WP.com application
+ * @property scope The scope of the Quick Editor
+ */
+public class GravatarQuickEditorParams private constructor(
+    public val appName: String,
+    public val oAuthParams: OAuthParams,
```

Comment:
🤔 I think we shouldn't include the `OAuthParams` inside `GravatarQuickEditorParams`. In my mind, being totally strict, the `QuickEditor` doesn't need the OAuthParams. It needs a token to connect with the backend. For that reason, I would consider keeping it outside this class. 

That way, we can have two `GravatarQuickEditorPage` composables. One when we already have the token, for example, if we finally decide to save it locally or the app has already the token:

```
@Composable
internal fun GravatarQuickEditorPage(
    gravatarQuickEditorParams: GravatarQuickEditorParams,
    oauthToken: String
    onAuthError: () -> Unit = {},
) {
```

And another option if want to go through the OAuth flow:

```
@Composable
internal fun GravatarQuickEditorPage(
    gravatarQuickEditorParams: GravatarQuickEditorParams,
    oAuthParams: OAuthParams,
    onAuthError: () -> Unit = {},
) {
```

Reply:
Yeah, I can extract it. The main idea was to have fewer params (with callbacks this will grow quite a bit) and to have one object for the quick editor configuration.

>One when we already have the token, for example, if we finally decide to save it locally or the app has already the token:

Whether we store it or not the token might still expire, and in such a case, we won't be able to handle this. As long as we are handling the token ourselves (which is what we will do I believe) we need that OAuth config.

I guess one specific case is the Jetpack app, this one will have the token already, right?

Reply:
>I guess one specific case is the Jetpack app, this one will have the token already, right?

I think for this, since we won't have to launch the OAuth at all we would create a new bottom sheet that will open the Avatar editor directly without all the OAuth -> Editor navigation logic. So we would pretty much skip the `GravatarQuickEditorPage` completely. Thoughts?

Reply:
> So we would pretty much skip the GravatarQuickEditorPage completely. 

Yeah, we will probably open the already authorized content of the `GravatarQuickEditorPage` that will be extracted in a different composable.